Sources/Controllers/ALKConversationListTableViewController.swift

```diff
@@ -119,7 +119,114 @@ public class ALKConversationListTableViewController: UITableViewController, Loca
             tableView.sectionHeaderTopPadding = 0
         }
         tableView.estimatedRowHeight = 0
+        guard configuration.enableDeleteConversationOnLongpress else{ return }
+        let longPress = UILongPressGestureRecognizer(target: self, action: #selector(longPress))
+        tableView.addGestureRecognizer(longPress)
+    }
+    
+    @objc func longPress(sender: UILongPressGestureRecognizer) {
+        if sender.state == UIGestureRecognizer.State.began {
+            let touchPoint = sender.location(in: tableView)
+            if let indexPath = tableView.indexPathForRow(at: touchPoint) {
+                if viewModel.chatFor(indexPath: indexPath) != nil, let conversation = viewModel.getChatList()[indexPath.row] as? ALMessage {
+                    showDeleteAlert(conversation: conversation)
+                }
+            }
+        }
+    }
+    
+    func showDeleteAlert(conversation: ALMessage) {
+        let title = localizedString(
+            forKey: "DeleteConversationTitle",
+            withDefaultValue: SystemMessage.DeleteConversationPopup.title,
+            fileName: localizedStringFileName
+        )
+        
+        let warningMessage =  localizedString(
+            forKey: "DeleteConversationContent",
+            withDefaultValue: SystemMessage.DeleteConversationPopup.content,
+            fileName: localizedStringFileName
+        )
+        
+        let alert = UIAlertController(title: title, message:warningMessage, preferredStyle: .alert)
+
+        let cancelButton = UIAlertAction(
+            title: localizedString(
+                forKey: "ButtonCancel",
+                withDefaultValue: SystemMessage.ButtonName.Cancel,
+                fileName: localizedStringFileName
+            ),
+            style: .cancel,
+            handler: nil
+        )
+        
+       let buttonTitle =  localizedString(
+            forKey: "DeleteButtonName",
+            withDefaultValue: "Delete",
+            fileName: localizedStringFileName
+        )
+        
+        let deleteButton = UIAlertAction(title: buttonTitle, style: .destructive, handler: { [weak self] _ in
+            guard let weakSelf = self, ALDataNetworkConnection.checkDataNetworkAvailable() else { return }
+            let alert = weakSelf.displayLoadingAlert(viewController: weakSelf)
+            let messageService = ALMessageService()
+            if conversation.isGroupChat {
+                messageService.deleteMessageThread(nil, orChannelKey: conversation.groupId, withCompletion: {
+                    _, error in
+                    alert.dismiss(animated: false)
+                    guard error == nil else {
+                        weakSelf.showAlertWithSingleAction(message: error.debugDescription)
```

Comment:
Directly showing the error may not be a good idea. Better to log the error and show a static failure message.

Reply:
done